src/server/game/Auction/AsyncAuctionOperation.cpp

```diff
@@ -0,0 +1,505 @@
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AsyncAuctionOperation.h"
+#include "AuctionHouseMgr.h"
+#include "AuctionHousePackets.h"
+#include "Creature.h"
+#include "DatabaseEnv.h"
+#include "GameTime.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+
+Player* AsyncAuctionOperation::GetPlayer() const
+{
+    return ObjectAccessor::FindPlayer(_playerGuid);
+}
+
+void SellItemTask::Execute()
+{
+    auto player{ GetPlayer() };
+    if (!player)
+        return;
+
+    auto session{ player->GetSession() };
+
+    if (_packet->ItemsCount > MAX_AUCTION_ITEMS)
+    {
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    for (uint32 i = 0; i < _packet->ItemsCount; ++i)
+    {
+        auto const& [itemGuid, itemCount] = _packet->Items[i];
+        if (!itemGuid || !itemCount || itemCount > 1000)
+        {
+            session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_ITEM_NOT_FOUND);
+            return;
+        }
+    }
+
+    if (!_packet->Bid || !_packet->ExpireTime)
+    {
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    if (_packet->Bid > MAX_MONEY_AMOUNT || _packet->Buyout > MAX_MONEY_AMOUNT)
+    {
+        LOG_ERROR("network", "WORLD: HandleAuctionSellItem - Player {} ({}) attempted to sell item with higher price than max gold amount.", player->GetName(), player->GetGUID().ToString());
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    Creature* creature = player->GetNPCIfCanInteractWith(_packet->Auctioneer, UNIT_NPC_FLAG_AUCTIONEER);
+    if (!creature)
+    {
+        LOG_ERROR("network", "WORLD: HandleAuctionSellItem - Unit ({}) not found or you can't interact with him.", _packet->Auctioneer.ToString());
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    AuctionHouseEntry const* auctionHouseEntry = AuctionHouseMgr::GetAuctionHouseEntry(creature->GetFaction());
+    if (!auctionHouseEntry)
+    {
+        LOG_ERROR("network", "WORLD: HandleAuctionSellItem - Unit ({}) has wrong faction.", _packet->Auctioneer.ToString());
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    Seconds depositTime{ Minutes{ _packet->ExpireTime } };
+    if (depositTime != 12h && depositTime != 1_days && depositTime != 2_days)
+    {
+        LOG_ERROR("network", "HandleAuctionSellItem: Incorrect deposit time: {}. Player info: {}", Acore::Time::ToTimeString(depositTime), session->GetPlayerInfo());
+        session->SendAuctionCommandResult(nullptr, AUCTION_SELL_ITEM, ERR_AUCTION_DATABASE_ERROR);
+        return;
+    }
+
+    if (player->HasUnitState(UNIT_STATE_DIED))
```

Comment:
This task (SellItemTask) will run in separate thread, right? And in this thread you are interacting with a player object. Can we have a race condition in this case?